src/box/space.h

```diff
@@ -284,6 +284,8 @@ struct space {
 	 * this object in sync. For more information see #9120.
 	 */
 	int lua_ref;
+	/** The effective state of synchronous replication for this space. */
+	bool is_sync;
```

Comment:
Let's try to find another name for this variable.
In Lua it's good (`state.is_sync`), but here it's too easy to confuse with `space_def::is_sync`.

I can't think of a decent name, so maybe we should hide it into a substruct:
```c
struct {
    bool is_sync;
} state;
```

Then we'll have `space->def->is_sync` (the desired state set by the user) and `space->state.is_sync` (the actual state). It'll also mirror the Lua layout you came up with quite nicely.

Reply:
Fixed.